### PR TITLE
start transition of CI according to apache infra reqs

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description
This PR transitions from travis to gha. it is trying to follow https://docs.github.com/en/actions/migrating-to-github-actions/migrating-from-travis-ci-to-github-actions

An alternative approach would be to use: https://github.com/marketplace/actions/run-travis-yml (parallel PR?)

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

so far only in the original branch on the PR repo. It can be replicated to your own.

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@potiuk @rohityadavcloud I moved this to an apache branch so all that have interest can add to it.

File: .github/workflows/smokeAndComponents.yml [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+                    component/test_vpc_routers
+                    component/test_vpn_users" ]
+
+    steps:
+      - name: 'Set up Python ${{ matrix.python-version }}'
+        uses: actions/setup-python@v2
+        with:
+          python-version: '${{ matrix.python-version }}'
+      - name: 'Set up Node.js ${{ matrix.node-version }}'
+        uses: actions/setup-node@v1
```
Comment (jbampton) [inline .github/workflows/smokeAndComponents.yml:173]:
```suggestion
        uses: actions/setup-node@v3
```
https://github.com/actions/setup-node

File: .github/workflows/smokeAndComponents.yml [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+                    component/test_vpc_distributed_routing_offering
+                    component/test_vpc_network",
+                  "component/test_vpc_network_lbrules
+                    component/test_vpc_offerings
+                    component/test_vpc_routers
+                    component/test_vpn_users" ]
+
+    steps:
+      - name: 'Set up Python ${{ matrix.python-version }}'
+        uses: actions/setup-python@v2
```
Comment (jbampton) [inline .github/workflows/smokeAndComponents.yml:169]:
```suggestion
        uses: actions/setup-python@v4
```

https://github.com/actions/setup-python

**Review by jbampton — COMMENTED** [review]
You can use Dependabot to maintain your "GitHub Actions"

Customize how Dependabot maintains your repositories:

https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file

**Comment by DaanHoogland** [comment]
@assignUser Can you have a brief review of this? Maybe^H^H^H^H^HProbably I am missing something obvious...

**Comment by DaanHoogland** [comment]
thanlks @assignUser , not please the @rohityadavcloud has taken this further by now in #6956, maybe you can give us some advice there?

**Comment by DaanHoogland** [comment]
closing in favour of #6956

**Comment by DaanHoogland** [comment]
closing in favour of #6956

**Pull request closed without merging**